### DISCUSSION TITLE
Is it possible to open c/c++ or any text file with readonly mode in thiea

### DISCUSSION THREAD

**Discussion by wss29** [eclipse-theia/theia#14083] [Q&A] [answered]
Hi
   Is it possible to open c/c++ or any text file with read-only mode in thiea?

**Comment by msujew** [comment, marked as answer]
Hey @wss29,

yes, see:

https://github.com/eclipse-theia/theia/blob/754eb84caecc8e318def10c6b96e4d11ba7081b9/packages/filesystem/src/browser/file-resource.ts#L352-L356

See also this command in our example app that makes the whole file system readonly:

https://github.com/eclipse-theia/theia/blob/754eb84caecc8e318def10c6b96e4d11ba7081b9/examples/api-samples/src/browser/file-system/sample-file-system-capabilities.ts#L30-L42
